### PR TITLE
Add HAProxy remote access example

### PR DESCRIPTION
**Pull request by mikeodr**
**Description:**

Created an HAProxy remote access proxy documentation and
example configuration.

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_docs/ecosystem/haproxy.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	mode http
+	option forwardfor
+	http-request add-header X-Forwarded-Proto https
+	http-request add-header X-Forwarded-Port 443
+```
+
+### {% linkable_title Forward Ports %}
+
+Forward ports 443 and (optionally) 80 to your server on your router. Do not forward port 8123.
```
Comment (DubhAd) [inline source/_docs/ecosystem/haproxy.markdown]: Why not forward port 8123? This should be explained?

File: source/_docs/ecosystem/haproxy.markdown [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Replace 443 with whatever port you chose to bind to in the config if different.
+
+### {% linkable_title Configure Home Assistant HTTP Component %}
+
+In your `configuration.yaml` file, edit the [http component](https://www.home-assistant.io/components/http/).
+
+```text
+http:
+  server_host: 127.0.0.1
```
Comment (DubhAd) [inline source/_docs/ecosystem/haproxy.markdown]: This guide seems to assume HAProxy is going on the same host as Home Assistant, but it never actually says so, which will trip people up.

**Review by DubhAd — DISMISSED** [review]

**Comment by mikeodr** [comment]
@DubhAd updated.

**Review by klaasnicolaas — DISMISSED** [review]

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @mikeodr! 👍

**Pull request merged**